scenes/Ships/Ship_controls.gd

```diff
@@ -69,22 +70,26 @@ func _physics_process(delta: float) -> void:
 	if _accelerating:
 		velocity = velocity.move_toward(Vector2(1, 0).rotated(rotation + PI / 2) * max_speed, acceleration * delta)
 	
-	if _apply_collision_knockback_damage():
-		velocity = velocity * -1
-	
 	if GameState.player.hull_health <= 0: 
 		queue_free()
 
-
 	GameState.player.position = self.position
 	
 	set_velocity(velocity)
-	set_floor_stop_on_slope_enabled(false)
-	set_max_slides(4)
-	set_floor_max_angle(PI/4)
-	# TODOConverter40 infinite_inertia were removed in Godot 4.0 - previous value `false`
-	move_and_slide()
-	velocity = velocity
+	var collision = move_and_collide(velocity * delta)
+	if collision != null:
+		_handle_collision(collision)
+
+func _handle_collision(collision: KinematicCollision2D):
+	var collider = collision.get_collider()
+	if collider is Asteroid and Time.get_unix_time_from_system() - _last_asteroid_collision > 0.5:
```

Comment:
If the ship is colliding with an asteroid and we haven't already collided with one in the last 0.5s then apply an impact to the asteroid at the point where the collision occurred.